### PR TITLE
LPS-119068 [Page Audit] Fix Performance Issue

### PR DESCRIPTION
**Pull request by 4lejandrito**
**Description**

The QA team run performance tests within Liferay CE version. We added a new sidebar panel for Page Audit functionality and this introduces about 5% regression to portal because the LayoutReportsPortlet gets rendered even if the panel is not expanded. 

**Changes within this pull request:**

1. Remove LayoutReportsPortlet rendering when loading the page.
2. Remove the AJAX call when clicking (thought icon data-url).
3. Remove JSPs and move sidebar markup to the JSPWriter in the menu entry file.
4. Render the React component in the menu entry file.
5. Subscribe to onmouseenter (with mouse) and onfocus (with keyboard) to call the endpoint to display the Page Audit data. This data will be called once.
6. If the panel remains open and the user refresh the page, the panel will be also loaded (React code useEffectLayout).

### CODE REVIEW

**Comment by dgarciasarai** [comment]
Reviewing 😃

**Comment by dgarciasarai** [comment]
ci:test:sf

File: modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js [hunk 1]
```diff
@@ -12,33 +12,211 @@
...
+
+	const [state, dispatch] = useReducer(dataReducer, initialState);
+
+	const safeDispatch = (action) => {
+		if (isMounted()) {
+			dispatch(action);
+		}
+	};
+
+	const [isLoaded, setIsLoaded] = useState(false);
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js:82]: This state is initialize in every render, so `getData` is called again when `mouseenter` or `focus` happen.

File: modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js [hunk 2]
```diff
@@ -12,33 +12,211 @@
...
+
+		Liferay.once('screenLoad', () => {
+			Liferay.SideNavigation.destroy(layoutReportsPanelToggle);
+		});
+	}, [portletNamespace]);
+
+	useLayoutEffect(() => {
+		if (isPanelStateOpen) {
+			getData(layoutReportsDataURL);
+			setIsLoaded(true);
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js:137]: An this is not doing anything

File: modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js [hunk 3]
```diff
@@ -12,33 +12,211 @@
...
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [layoutReportsDataURL]);
+
+	const handlePageAuditData = useCallback(() => {
+		if (!isLoaded) {
+			getData(layoutReportsDataURL);
+			setIsLoaded(true);
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js:145]: Also here

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Hey @4lejandrito, I think the frontend part is done. Although, we will send these changes to the Frontend Infra team to review once the backend part is finish and CI tests are passing successfully. Thanks!

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
@dgarciasarai 🥳

**Comment by GinsonRen** [comment]
@4lejandrito @dantewang Here is the diff result:

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of Logout | Session Count Change （%）
-- | -- | -- | -- | -- | --
BaseLine | a7701997 | 29060 | 567 | 174 | -
PR-1070 | - | 29145 | 570 | 176 | 0.29%

Since page audit already be disabled on master head (a7701997) by default, I deploy the `com.liferay.layout.reports.web.internal.configuration.LayoutReportsGooglePageSpeedConfiguration.config`  and set `enabled=B"true"` into the config.

**Comment by dantewang** [comment]
@4lejandrito The results can be considered identical. It's good to go from our side, thank you!

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:reevaluate:1077401_7049

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by CarlosBrichete** [comment]
Just started reviewing :)

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
Sent in paralell directly to Brian: https://github.com/brianchandotcom/liferay-portal/pull/100981.

**Comment by 4lejandrito** [comment]
Merged

**Comment by dantewang** [comment]
@4lejandrito It's merged but then reverted...

**Comment by dgarciasarai** [comment]
Commits are in Brian's repo again. See https://github.com/brianchandotcom/liferay-portal/commits/1e25d226a8c903d3579c88322ed6aadd93d0c215. So this pull is merged and closed 👍 
Thanks!

**Pull request closed without merging**